zds/utils/management/commands/migrate_from_gravatar.py

```diff
@@ -0,0 +1,20 @@
+from hashlib import md5
+
+import requests
+from django.core.management.base import BaseCommand
+
+from zds.member.models import Profile
+
+
+class Command(BaseCommand):
+    help = "Migrate from Gravatar"
+
+    def handle(self, *args, **options):
+        for profile in Profile.objects.select_related("user").filter(avatar_url__isnull=True).iterator():
+            hash = md5(profile.user.email.lower().encode("utf-8")).hexdigest()
+            gravatar_url = f"https://secure.gravatar.com/avatar/{hash}"
+            r = requests.get(f"{gravatar_url}?d=404")
```

Comment:
Je suis en train de me dire qu'on ne peut pas exécuter une requête comme ça dans une boucle, on va se faire ban pour débit de requêtes trop élevé...

D'après [la doc](https://docs.gravatar.com/api/avatars/images/) qui renvoie vers [une page de Wordpress](https://developer.wordpress.com/guidelines/) : "Use of our API is free. We only ask that you not abuse it by making excessive calls through your apps. ". A priori pas de rate limit, mais sait-on jamais...

Je viens de regarder, sur la bêta la requête exécutée renvoie 25298 enregistrements. À raison de 1 requête par seconde, ça prendrait 7h... Ça peut très bien se laisser tourner en tâche de fond. Mais ça veut dire qu'il faut afficher la progression, pour savoir où on en est.

Reply:
Punaise, je sais que j'y ai déjà pensé mais jamais au moment de coder xD Bref, j'ai ajouté le délai d'une seconde entre chaque requête et un petit compteur pour la progression !